### PR TITLE
Make hyspometric interpolation methods more gap-friendly

### PR DESCRIPTION
**Pull request by adehecq**
These are some fix I had to make to run the local hypsometric interpolation on a ddem with very large gaps, for example with outlines falling completely outside of valid range.
@erikmannerfelt please check if what I've made makes sense and we can discuss how to better account for such cases.

### CODE REVIEW

File: xdem/ddem.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def interpolate(self, method: str = "linear",
                         interpolated_ddem,
                         reference_elevation.data,
                         mask=feature_mask
-                    ).data
+                    )#.data   # This causes an issue at l 172: NotImplementedError: sub-views are not implemented
```
Comment (erikmannerfelt) [inline xdem/ddem.py:160]: This entire statement should probably be encased in a `np.asarray(...)` instead. `xdem.volume.hypsometric_interpolation()` should return a masked_array (but we want ndarrays in this case), which is why I took the data property. 

Unfortunately, `np.ndarray.data` is something completely different from `np.ma.masked_array.data`!!

**Review by erikmannerfelt — COMMENTED** [review]
Great! I have one suggestion to be a bit more future-proof, but otherwise it looks good.

**Comment by adehecq** [comment]
Solves #84

**Review by erikmannerfelt — APPROVED** [review]
Great!!

**Pull request merged**